@xen-orchestra/backups/_cleanVm.js

```diff
@@ -528,6 +532,12 @@ exports.cleanVm = async function cleanVm(
     }
   })
 
+  // purge cache if a metadata file has been deleted
+  if (mustInvalidateCache) {
+    const vmUuid = basename(vmDir)
```

Comment:
I'm not fond of this as it makes assumptions about the path (assumptions that can be false for instance when running clean VM manually).

What do you think of adding a protected method to the remote adapter that would take a dir path instead of VM UUID and make `invalidateVmBackupListCache` use it?

Reply:
Or you can even delete it manually.